sdk/monitor/Azure.Monitor.Ingestion/src/LogsIngestionClient.cs

```diff
@@ -266,141 +264,147 @@ public virtual async Task<Response<UploadLogsResult>> UploadAsync<T>(string rule
+            return runningTasks.Select(_ => _.CurrentTask).Last().Result; //204 - response of last batch with header
         }
 
-        private async Task<BatchUpload> CommitBatchListSyncOrAsync<T>(BatchedLogs<T> batch, string ruleId, string streamName, bool async, CancellationToken cancellationToken)
+        private async Task<Response> UploadBatchListSyncOrAsync<T>(BatchedLogs<T> batch, string ruleId, string streamName, bool async, CancellationToken cancellationToken)
         {
-            UploadLogsError error = null;
-            RequestContext requestContext = GenerateRequestContext(cancellationToken);
             Response response = null;
 
-            using HttpMessage message = CreateUploadRequest(ruleId, streamName, batch.LogsData, "gzip", requestContext);
+            using HttpMessage message = CreateUploadRequest(ruleId, streamName, batch.LogsData, Compression, null);
 
             if (async)
             {
-                response = await _pipeline.ProcessMessageAsync(message, requestContext, cancellationToken).ConfigureAwait(false);
+                response = await _pipeline.ProcessMessageAsync(message, null, cancellationToken).ConfigureAwait(false);
             }
             else
             {
-                response = _pipeline.ProcessMessage(message, requestContext, cancellationToken);
+                response = _pipeline.ProcessMessage(message, null, cancellationToken);
             }
 
-            if (response.Status != 204) // if any error is thrown log it
-            {
-                RequestFailedException requestFailedException = new RequestFailedException(response);
-                ResponseError responseError = new ResponseError(requestFailedException.ErrorCode, requestFailedException.Message);
-                error = new UploadLogsError(responseError, (IReadOnlyList<BinaryData>)batch.LogsList);
-            }
-            return new BatchUpload(response, error);
+            return response;
         }
 
-        private static RequestContext GenerateRequestContext(CancellationToken cancellationToken)
+        private static List<Exception> AddException(List<Exception> exceptions, Exception ex)
         {
-            var requestContext = new RequestContext() { CancellationToken = cancellationToken };
-            requestContext.AddClassifier(500, false);
-            requestContext.AddClassifier(403, false);
-            requestContext.AddClassifier(413, false);
-            requestContext.AddClassifier(429, false);
-            requestContext.AddClassifier(503, false);
-            return requestContext;
+            exceptions ??= new List<Exception>();
+            exceptions.Add(ex);
+            return exceptions;
         }
 
-        private static UploadLogsStatus GetStatus<T>(IEnumerable<T> logEntries, List<UploadLogsError> errors)
+        private static void ThrowException<T>(IEnumerable<T> logs, DiagnosticScope scope, List<Exception> exceptions, int logsFailed)
```

Comment:
Why is this generic?

Reply:
the method didn't have anything to be returned but I deleted it